### PR TITLE
[Audio 10/10] Loose ends

### PR DESCRIPTION
**Pull request by Thar0**
The final PR in the series of initial audio work. With this PR everything I had worked on in my branch is merged, but new improvements may still happen in the future.
- Adds `afile_sizes` to generate sizes headers for Soundfonts and Sequences for `audio/session_config.c`
- Adds an overview README for tools/audio
- Tools cleanup and further documentation
- Renames some internal .note sections from `.name` and `.fonts` to `.note.name` and `.note.fonts`
- Samplebank and Soundfont xml documentation

I originally planned to also include a doc about how VADPCM works but writing it was taking too long so I didn't include it here.

### CODE REVIEW

**Review by fig02 — APPROVED** [review]
mostly just read docs and didnt look at code changes too deeply

File: docs/audio/Samplebank_XML.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    Name="<C Indentifier>"
+    Index="<uint>"
+    Medium="<Medium>"
+    CachePolicy="<CachePolicy>"
+    BufferBug="[bool]"
+>
+```
+Begins a new samplebank.
+
+**Properties**
```
Comment (cadmic) [inline docs/audio/Samplebank_XML.md:18]: technically in XML parlance these are called "attributes", and what's called "Sub-Attributes" below are "tags" (maybe "child tags" or "children"?). I think using the word "properties" is probably fine, but using the word "attribute" for tags could be confusing

(ditto with other XML docs)

File: docs/audio/Samplebank_XML.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        Name="<C Identifier>"
+        Path="<Path>"
+    />
+    ```
+    Adds a binary blob to the samplebank. Intended for matching only when data cannot be identified.
+
+    **Properties**
+
+    - **Name**: Name of this blob. Must be a valid C language identifier.
+    - **Path**: Path to binary file, relative to the project root (typically in `assets/samples/`)
```
Comment (cadmic) [inline docs/audio/Samplebank_XML.md:70]: should be `$(BUILD_DIR)/assets/audio/samples/`?

File: docs/audio/Samplebank_XML.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+>
+```
+Begins a new samplebank.
+
+**Properties**
+
+- **Name**: The name of the samplebank.
+- **Index**: The index of the samplebank for the samplebank table. Must be a unique index for all samplebanks and pointers.
+- **Medium**: The storage medium, from the SampleMedium enum.
+- **CachePolicy**: The cache policy, from the AudioCacheLoadType enum.
```
Comment (cadmic) [inline docs/audio/Samplebank_XML.md:23]: super nit: in the soundfont XML you put `SampleMedium` and `AudioCacheLoadType` in monospace font

File: docs/audio/Soundfont_XML.md [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+            Name="<C Identifier>"
+            Release="<u8>"
+        >
+        ```
+        Starts a new envelope.
+
+        **Properties**
+
+        - **Name**: Unique name for this envelope. Must be a valid C identifier.
+        - **Release**: Release rate index for this envelope
```
Comment (cadmic) [inline docs/audio/Soundfont_XML.md:63]: what does this index into exactly? I tried searching for "release rate" but didn't find much

File: docs/audio/Soundfont_XML.md [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+
+    **Sub-Attributes**
+
+    -
+        ```xml
+        <Effect
+            Name="<C Identifier>"
+            Sample="<Sample Name>"
+            SampleRate="[Sample Rate]"
+            BaseNote="[Note Number]"
```
Comment (cadmic) [inline docs/audio/Soundfont_XML.md:184]: Define "Note Number"? If I didn't know better I'd attempt to put "C4" or something here

**Review by cadmic — APPROVED** [review]
I only skimmed the docs, and did not try to use them to create my own samplebank or soundfont

File: docs/audio/Soundfont_XML.md (hunk 4 above)
Comment (Thar0) [reply docs/audio/Soundfont_XML.md]: It indexes `gAudioCtx.adsrDecayTable`, I mentioned this but didn't add much explanation for it, I think the code surrounding this probably needs some docs before reviewing again how it's presented here

File: docs/audio/Soundfont_XML.md (hunk 5 above)
Comment (Thar0) [reply docs/audio/Soundfont_XML.md]: C4 would actually be correct, I've changed it to say `Note Name` and added a brief explanation of the field at the top of the file.

**Pull request merged**